src/format/strftime.rs

```diff
@@ -38,7 +38,7 @@ The following specifiers are available both to formatting and parsing.
 | `%j`  | `189`    | Day of the year (001--366), zero-padded to 3 digits.                       |
 |       |          |                                                                            |
 | `%D`  | `07/08/01`    | Month-day-year format. Same as `%m/%d/%y`.                            |
-| `%x`  | `07/08/01`    | Same as `%D`.                                                         |
+| `%x`  | `07/08/01`    | Locale's date representation (e.g., 12/31/99).                        |
```

Comment:
Oh darn why did we ever support this incorrectly 😢 

For all three of the ones that you changed the output of, I think we need to add a new format specifier `%#x` (and below `%#X` `%#c`) for the new Localized format, see e.g. `z` for the way that this is handled, it shouldn't be much more code:

https://github.com/chronotope/chrono/blob/130253ef2b914d300c1a0c6cff36cd3a5d846767/src/format/strftime.rs#L306-L312

Reply:
:scream: this would make things incompatible with glibc's format. I don't think that's a big deal though... :thinking: 

(Note that if the feature "locales" is not used, the original behavior is kept. I don't know if that makes it more okay with the compatibility you seek)

Reply:
Oh that's a great point!

Yes, could you just change the documentation string to say something like:

> With the `locales` feature this is the Locale's date representation, otherwise same as...

Reply:
done

Reply:
Oh! I just saw your message. Nevermind then, I will revert to the other state and update the doc as you said (I like it better personally!)

Reply:
done